import-scripts/pipelines_eks/automation-environment.sh

```diff
@@ -11,7 +11,7 @@ export PYTHON3_BINARY=/usr/bin/python3
 export MAVEN_BINARY=/opt/apache-maven-3.8.1/bin/mvn
 export HG_BINARY=/usr/bin/hg
 export GIT_BINARY=/usr/bin/git
-export PATH=/opt:/usr/bin:/data/tools/python2.7/bin:/data/tools/python3.4.10/bin:$PATH
+export PATH=$(bash --login -c 'echo $PATH')
```

Comment:
not familiar with this but off google looks like it loads in what's in bash_profile?

Reply:
bash can start up in a few different modes, based on whether it is connected to an interactive terminal (like when we are manually logged in) or not. So on startup, bash is considered one of {interactive, non-interactive} and also one of {login, non-login}. When a bash process is launched by crontab it is considered non-interactive and non-login and the normal bash_profile startup scripts are not executed. The only PATH which is set is a very rudimentary default -- it is only PATH="/usr/bin:/bin". However, you can tell bash to start up as a login shell with the command line option "--login" ... and then the normal bash_profile initialization happens. So this code launches a new bash subshell as a login shell and extracts the initialized PATH environment variable from the subshell. It then sets the top level shell to have the same PATH.